zebra-chain/src/sapling/commitment.rs

```diff
@@ -164,6 +188,44 @@ impl From<ValueCommitment> for [u8; 32] {
     }
 }
 
+impl std::ops::Sub<ValueCommitment> for ValueCommitment {
+    type Output = Self;
+
+    fn sub(self, rhs: ValueCommitment) -> Self::Output {
+        ValueCommitment((self.0.to_extended() - rhs.0.to_extended()).into())
+    }
+}
+
+impl std::ops::SubAssign<ValueCommitment> for ValueCommitment {
+    fn sub_assign(&mut self, rhs: ValueCommitment) {
+        *self = *self - rhs;
+    }
+}
+
+impl std::iter::Sum for ValueCommitment {
+    fn sum<I>(iter: I) -> Self
+    where
+        I: Iterator<Item = Self>,
+    {
+        iter.fold(
+            ValueCommitment(jubjub::AffinePoint::identity()),
+            std::ops::Add::add,
+        )
+    }
+}
+
+impl<'a> std::iter::Sum<&'a ValueCommitment> for ValueCommitment {
+    fn sum<I>(iter: I) -> Self
+    where
+        I: Iterator<Item = &'a Self>,
+    {
+        iter.fold(
+            ValueCommitment(jubjub::AffinePoint::identity()),
+            std::ops::Add::add,
+        )
+    }
+}
+
```

Comment:
Similar to the `Add` question, can we re-implement this function using the non-reference `Sum for ValueCommitment`?

Reply:
I'm not quite sure the best way to do this, but I don't think I necessarily need this one so I'm removing it.